? qa/rtl/oustring/signalfile_1180535446.txt
? qa/rtl/oustring/signalfile_1180535459.txt
Index: qa/rtl/oustring/rtl_OUString2.cxx
===================================================================
RCS file: /cvs/porting/sal/qa/rtl/oustring/rtl_OUString2.cxx,v
retrieving revision 1.9
diff -u -p -u -r1.9 rtl_OUString2.cxx
--- qa/rtl/oustring/rtl_OUString2.cxx	3 Apr 2007 14:18:04 -0000	1.9
+++ qa/rtl/oustring/rtl_OUString2.cxx	30 May 2007 14:31:52 -0000
@@ -998,6 +998,10 @@ public:
 
     void intern()
     {
+        // The empty string is 'static' a special case ...
+        rtl::OUString aEmpty = rtl::OUString().intern();
+        rtl::OUString aEmpty2 = rtl::OUString::intern( RTL_CONSTASCII_USTRINGPARAM( "" ) );
+
         ::rtl::OUString aFoo( RTL_CONSTASCII_USTRINGPARAM("foo") );
         ::rtl::OUString aFooIntern = aFoo.intern();
         CPPUNIT_ASSERT_MESSAGE("string contents", aFooIntern.equalsAscii("foo"));
Index: rtl/source/hash.cxx
===================================================================
RCS file: /cvs/porting/sal/rtl/source/hash.cxx,v
retrieving revision 1.3
diff -u -p -u -r1.3 hash.cxx
--- rtl/source/hash.cxx	19 Apr 2007 11:56:36 -0000	1.3
+++ rtl/source/hash.cxx	30 May 2007 14:31:52 -0000
@@ -109,7 +109,8 @@ rtl_str_hash_intern (StringHashTable   *
             return NULL;
     }
 
-    pString->refCount |= SAL_STRING_INTERN_FLAG;
+    if (!SAL_STRING_IS_STATIC (pString))
+        pString->refCount |= SAL_STRING_INTERN_FLAG;
     pHash->insert(pString);
 
     return pString;
